### PR TITLE
Fix `BadImport` Error Prone violations

### PR DESCRIPTION
**Pull request by basil**
The [`BadImport`](https://errorprone.info/bugpattern/BadImport) Error Prone check comes with the following rationale:

>  Importing nested classes/static methods/static fields with commonly-used names can make code harder to read, because it may not be clear from the context exactly which type is being referred to. Qualifying the name with that of the containing class can make the code clearer. 

In this PR I disambiguate any ambiguous imports by qualifying them with the name of the containing class.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/model/DownloadService.java
Comment (basil) [inline core/src/main/java/hudson/model/DownloadService.java:388]: Not to be confused with `javax.tools.Diagnostic.Kind`.

File: core/src/main/java/hudson/model/FullDuplexHttpChannel.java
Comment (basil) [inline core/src/main/java/hudson/model/FullDuplexHttpChannel.java:55]: Not to be confused with `hudson.model.Node.Mode`.

File: core/src/main/java/hudson/model/Queue.java
Comment (basil) [inline core/src/main/java/hudson/model/Queue.java:1782]: Not to be confused with `hudson.util.CopyOnWriteMap.Hash`.

File: core/src/main/java/hudson/model/UpdateSite.java
Comment (basil) [inline core/src/main/java/hudson/model/UpdateSite.java:241]: Not to be confused with `javax.tools.Diagnostic.Kind`.

File: core/src/main/java/hudson/model/UsageStatistics.java
Comment (basil) [inline core/src/main/java/hudson/model/UsageStatistics.java:146]: Not to be confused with a variety of other `DescriptorImpl`s.

File: core/src/main/java/jenkins/PluginSubtypeMarker.java
Comment (basil) [inline core/src/main/java/jenkins/PluginSubtypeMarker.java:72]: Not to be confused with `hudson.util.FormValidation.Kind`.

File: core/src/main/java/jenkins/security/UserDetailsCache.java
Comment (basil) [inline core/src/main/java/jenkins/security/UserDetailsCache.java:74]: Not to be confused with a variety of other `newBuilder`s.

File: core/src/test/java/hudson/util/ConsistentHashTest.java
Comment (basil) [inline core/src/test/java/hudson/util/ConsistentHashTest.java:188]: Not to be confused with `hudson.util.ConsistentHash.Hash`.

File: test/src/test/java/hudson/cli/ComputerStateTest.java
Comment (basil) [inline test/src/test/java/hudson/cli/ComputerStateTest.java:104]: Not to be confused with `hudson.model.Cause.UserCause`.

File: test/src/test/java/hudson/cli/ComputerStateTest.java
Comment (basil) [inline test/src/test/java/hudson/cli/ComputerStateTest.java:123]: Not to be confused with `hudson.model.Cause.UserCause`.

File: test/src/test/java/hudson/model/NodeTest.java
Comment (basil) [inline test/src/test/java/hudson/model/NodeTest.java:119]: Not to be confused with `hudson.model.Cause.UserCause`.

File: test/src/test/java/hudson/model/NodeTest.java
Comment (basil) [inline test/src/test/java/hudson/model/NodeTest.java:127]: Not to be confused with `hudson.model.Cause.UserCause`.

File: test/src/test/java/hudson/model/NodeTest.java
Comment (basil) [inline test/src/test/java/hudson/model/NodeTest.java:144]: Not to be confused with `hudson.model.Cause.UserCause`.

File: test/src/test/java/hudson/model/NodeTest.java
Comment (basil) [inline test/src/test/java/hudson/model/NodeTest.java:153]: Not to be confused with `hudson.model.Cause.UserCause`.

File: test/src/test/java/hudson/model/SlaveTest.java
Comment (basil) [inline test/src/test/java/hudson/model/SlaveTest.java:125]: Not to be confused with `javax.tools.Diagnostic.Kind`.

File: test/src/test/java/hudson/tasks/MavenTest.java
Comment (basil) [inline test/src/test/java/hudson/tasks/MavenTest.java:185]: Not to be confused with a variety of other `DescriptorImpl`s.

File: test/src/test/java/hudson/triggers/SCMTriggerTest.java
Comment (basil) [inline test/src/test/java/hudson/triggers/SCMTriggerTest.java:158]: Not to be confused with `hudson.slaves.OfflineCause.UserCause`.

File: test/src/test/java/hudson/triggers/SCMTriggerTest.java
Comment (basil) [inline test/src/test/java/hudson/triggers/SCMTriggerTest.java:162]: Not to be confused with `hudson.slaves.OfflineCause.UserCause`.

File: test/src/test/java/hudson/util/FormFieldValidatorTest.java
Comment (basil) [inline test/src/test/java/hudson/util/FormFieldValidatorTest.java:77]: Not to be confused with a variety of other `DescriptorImpl`s.

File: core/src/main/java/jenkins/model/lazy/BuildReferenceMapAdapter.java
Comment (basil) [inline core/src/main/java/jenkins/model/lazy/BuildReferenceMapAdapter.java:395]: The only actual change of substance in this PR. `groovy.util.MapEntry` seemed like an odd implementation of a map entry, as it is meant for use by Groovy and just happens to be on the classpath. This is also the only import from `groovy.util` in `src/main/` and seemed like an oversight. The implementation in `AbstractMap.SimpleEntry` is essentially identical, and it seems better to rely on the Java Platform for this than on Groovy. Note that I intentionally did _not_ choose `AbstractMap.SimpleImmutableEntry`, since the Groovy version is mutable and I did not want to change the behavior here.

File: core/src/main/java/hudson/scm/ChangeLogAnnotator.java
Comment (timja) [inline core/src/main/java/hudson/scm/ChangeLogAnnotator.java:34]: this causes javadoc to fail to build

> core/src/main/java/hudson/scm/ChangeLogAnnotator.java:77: error: reference not found
> will return the same string as {@link Entry#getMsgEscaped()}.

**Comment by uhafner** [comment]
How did you enable error prone in Jenkins builds? I tried that in my plugins without success. Or does this just work in Jenkins core but not in plugins?

File: core/src/main/java/hudson/scm/ChangeLogAnnotator.java
Comment (basil) [reply core/src/main/java/hudson/scm/ChangeLogAnnotator.java:34]: Fixed

**Comment by basil** [comment]
> How did you enable error prone in Jenkins builds? I tried that in my plugins without success. Or does this just work in Jenkins core but not in plugins?

Apply

```xml
diff --git a/pom.xml b/pom.xml
index eac25148d3..15ead5bc68 100644
--- a/pom.xml
+++ b/pom.xml
@@ -76,6 +76,8 @@ THE SOFTWARE.
     but this suppresses a warning from Maven, and as long as we don't do filtering we should be OK. -->
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
+    <javac.version>9+181-r4173-1</javac.version>
+
     <!-- configuration for patch tracker plugin  -->
     <project.patchManagement.system>github</project.patchManagement.system>
     <patch.request.organisation>jenkinsci</patch.request.organisation>
@@ -231,7 +233,17 @@ THE SOFTWARE.
                 ref: https://stackoverflow.com/questions/6770455/maven-compiling-package-info-java-to-package-info-class
               -->
               <compilerArg>-Xpkginfo:always</compilerArg>
+              <arg>-J-Xbootclasspath/p:${settings.localRepository}/com/google/errorprone/javac/${javac.version}/javac-${javac.version}.jar</arg>
+              <arg>-XDcompilePolicy=simple</arg>
+              <arg>-Xplugin:ErrorProne -Xep:ConstantOverflow:WARN -Xep:CollectionIncompatibleType:WARN -Xep:ChainingConstructorIgnoresParameter:WARN -Xep:BadAnnotationImplementation:WARN -Xep:BanSerializableRead:WARN -Xep:CheckReturnValue:WARN -Xep:GuardedBy:WARN -Xep:ReturnValueIgnored:WARN -Xep:MissingOverride:WARN -Xep:UnnecessaryParentheses:WARN -Xep:DefaultCharset:WARN -Xep:OverrideThrowableToString:WARN -Xep:ClassNewInstance:WARN -Xep:BadImport:WARN -Xep:JdkObsolete:WARN</arg>
             </compilerArgs>
+            <annotationProcessorPaths>
+              <path>
+                <groupId>com.google.errorprone</groupId>
+                <artifactId>error_prone_core</artifactId>
+                <version>2.7.1</version>
+              </path>
+            </annotationProcessorPaths>
           </configuration>
         </plugin>
         <plugin>
```

then run with `mvn clean verify -DskipTests -Dmaven.compiler.showWarnings=true`. See also [this page](https://errorprone.info/docs/patching) about automated patching. I've tried it, but it's a pain to get it to work with Maven multi-module builds and the automated patches typically need some polishing before they're ready to post as PRs.

**Review by timja — APPROVED** [review]

**Comment by uhafner** [comment]
> Apply [...] then run with mvn clean verify -DskipTests -Dmaven.compiler.showWarnings=true. 

I see, this is the same as I tried. So this seems to work for plain Java projects only (like core) but not with plugins.

**Review by StefanSpieker — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
